src/sdam/monitor.ts

```diff
@@ -16,6 +16,7 @@ import {
   ServerHeartbeatSucceededEvent,
   ServerHeartbeatFailedEvent
 } from './events';
+import { Server } from './server';
```

Comment:
actually now that I'm back here, don't we just need a `import type { Server }` here instead? I'm cautious about introducing unwanted cycles

Reply:
The `import type` syntax is a new insurance policy for what typescript already does, the generated code doesn't have an import cus this is only used as a type annotation. I think when we fix the prettier issue I think it was / eslint issue with this syntax I'd be interested to see if we can scan for these type only imports and make them explicit. I can change it to a type one now and do the eslint disable comment?

Reply:
Found this flag: `importsNotUsedAsValues`  that can be set to `'error'`, that will block code from compiling if there's an import that is type only but isn't annotated as such, I'll give it a go and report back

Reply:
Yea so this flag works as described, I'm going to add to the prettier ticket that we turn the flag on there and change `import`s to `import type`s where necessary.

Reply:
SGTM